offload/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -3561,6 +3564,28 @@ struct AMDGPUPluginTy final : public GenericPluginTy {
     return KernelAgents;
   }
 
+  Expected<MemoryInfoTy> getMemoryInfo(const void *TgtPtr) override {
```

Comment:
The code in `OffloadImpl` and the code in the NVidia case had/have `std::lock_guard` guarding. Is this not necessary here or has it been missed, or do I simply not see it?